src/pkg/services/m365/api/lists.go

```diff
@@ -504,6 +520,12 @@ func retrieveFieldData(orig models.FieldValueSetable, columnNames map[string]str
 	fields := models.NewFieldValueSet()
 
 	additionalData := setAdditionalDataByColumnNames(orig, columnNames)
+
+	if addressField, fieldName, ok := hasAddressFields(additionalData); ok {
+		concatenatedAddress := concatenateAddressFields(addressField)
+		additionalData[fieldName] = concatenatedAddress
```

Comment:
This is how a list with `location` columns look like:
![Screenshot from 2024-01-10 23-43-02](https://github.com/alcionai/corso/assets/48874082/4a647207-f5b0-4f08-ad23-fc524a24bea4)

with the above workaround, the restored list will look like:
![Screenshot from 2024-01-10 23-41-19](https://github.com/alcionai/corso/assets/48874082/10271bfe-75cb-4532-86ab-0efbbf09db05)

lists with `location` columns cannot be restored as the way it originally was created because of the issue described [here](https://github.com/microsoftgraph/msgraph-sdk-go/issues/638).